addons/addon-base-raas/packages/base-raas-services/lib/data-source/access-strategy/roles-only/helpers/entities/filesystem-role-methods.js

```diff
@@ -158,7 +162,43 @@ function toInlinePolicyDoc(fsRoleEntity = {}) {
     studyPolicy.addStudy(item);
   });
 
-  return studyPolicy.toPolicyDoc();
+  const vpcEpStatements = [];
+  const createConditionForVpce = (vpce, currentStudyIds) => {
+    const vpcEpStatement = {
+      Condition: {
+        StringNotEquals: {
+          'aws:SourceVpce': vpce,
+        },
+        StringLike: {
+          's3:prefix': _.map(currentStudyIds, studyId => {
+            const prefix = _.get(_.find(studyPolicy.studies, { id: studyId }), 'prefix');
+            // We need to account for the fact that a study might be the whole bucket, and when this
+            // is the case, the s3:prefix entry should be "*" and not "/*"
+            return prefix === '/' ? '*' : `${prefix}*`;
+          }),
+        },
+      },
+      Action: [
+        's3:Get*',
+        's3:List*',
+        's3:Put*',
+        's3:Delete*',
+        's3:AbortMultipartUpload',
+        's3:ListMultipartUploadParts',
+      ],
+      Resource: `arn:aws:s3:::${bucket}`,
+      Effect: 'Deny',
+      Sid: `AllowAccessFrom${vpce.split('-')[1]}`, // Managed Policy Sid cannot have anything but alphanumeric chars
```

Comment:
will there always be at least two elements from vpce after the split?

Reply:
Yes, VPC endpoint ID looks like `vpce-<randomString>`